ci-operator/step-registry/osd/create/create/osd-create-create-commands.sh

```diff
@@ -19,6 +19,7 @@ CLUSTER_MULTI_AZ=${CLUSTER_MULTI_AZ:-false}
 SSO_CLIENT_ID=$(read_profile_file "sso-client-id")
 SSO_CLIENT_SECRET=$(read_profile_file "sso-client-secret")
 OCM_TOKEN=$(read_profile_file "ocm-token")
+CLUSTER_TIMEOUT=${CLUSTER_TIMEOUT:-}
```

Comment:
`:-}` means setting default value for empty string.
But empty string will cause syntax failure in the `if`

```
bash-5.1 ~$ unset CLUSTER_TIMEOUT
bash-5.1 ~$ CLUSTER_TIMEOUT=${CLUSTER_TIMEOUT:-}
bash-5.1 ~$ start_time=$(date +"%s")
bash-5.1 ~$ (( $(date +"%s") - $start_time >= $CLUSTER_TIMEOUT ))
bash: ((: 1651045178 - 1651045174 >=  : syntax error: operand expected (error token is ">=  ")
```

Value should be always passed to the script  based on `osd-create-create-ref.yaml`
IMHO setting default value to empty string is not very useful we should rather rely on strict bash options `set -o nounset` to report unset variable

```
bash-5.1 ~$ cat test.sh 

#!/bin/bash
set -o nounset
set -o errexit
set -o pipefail

CLUSTER_TIMEOUT=${CLUSTER_TIMEOUT}

echo "Waiting for cluster ready..."
start_time=$(date +"%s")

if (( $(date +"%s") - $start_time >= $CLUSTER_TIMEOUT )); then
     echo "error: Timed out while waiting for cluster to be ready"
     exit 1
fi
bash-5.1 ~$ bash test.sh
test.sh: line 7: CLUSTER_TIMEOUT: unbound variable

bash-5.1 ~$ CLUSTER_TIMEOUT=10 bash test.sh
Waiting for cluster ready...
bash-5.1 ~$ echo $?
0
```

Reply:
thanks for such detailed explanation! It totally make sense, I fixed it based on your comment.